include/hmap/key-convert.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+/************************************************************************************
+ *
+ * Author: Thomas Dickerson
+ * Copyright: 2019 - 2020, Geopipe, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ************************************************************************************/
+
+#include "hmap.hpp"
+#include "dynamic-hmap.hpp"
+
+using detail::Key;
+using detail::KeyType;
+
+template<typename Value, char ...Cs>
+Key<Value> staticToDynamicKey(KeyType<Value, Cs...> key) {
+	return dK<Value>(key.c_str());
```

Comment:
Now that I think about it, we don't actually need to name the argument, and we can use `KeyType<Value, Cs...>::c_str()`, since `c_str()` is a static method, and hopefully the compiler will optimize out the variable entirely.